rest-service/manager_rest/rest/filters_utils.py

```diff
@@ -1,129 +1,138 @@
-    """Validate and parse a list of labels filters
-
-    :param labels_filters_list: A list of labels filters. Labels filters must
-           be one of: <key>=<value>, <key>=[<value1>,<value2>,...],
-           <key>!=<value>, <key>!=[<value1>,<value2>,...], <key> is null,
-           <key> is not null
-
-    :return The labels filters mapping dictionary with the following schema:
-            {NOT_EQUAL: {}, EQUAL: {}, IS_NULL: [], IS_NOT_NULL: []}
+        return filter_elem.value
+
+
+def parse_filter_rules(filter_rules: List[dict], resource_model: models):
+    """Validating the filter rules list.
+
+    :param filter_rules: A list of filter rules. A filter rule is a dictionary
+           of the following form:
+           {
+               key: <key>,
+               values: [<list of values>],
+               operator: <LabelsOperator> or <AttrsOperator>,
+               type: <FilterRuleType>
+            }
+    :param resource_model: models.Deployment or models.Blueprint
+    :return: Parsed filter rules list
     """
-    labels_filters_mapping = {}
-    for labels_filter in labels_filters_list:
-        if '!=' in labels_filter:
-            label_key, label_values = _parse_labels_filter(labels_filter, '!=')
-            # a!=b and a!=c <=> a!=[b, c]
-            labels_filters_mapping.setdefault(NOT_EQUAL, {}).setdefault(
-                label_key, []).extend(label_values)
-
-        elif '=' in labels_filter:
-            label_key, label_values = _parse_labels_filter(labels_filter, '=')
-            labels_filters_mapping.setdefault(EQUAL, {})
-            if label_key in labels_filters_mapping[EQUAL]:
-                raise manager_exceptions.BadParametersError(
-                    'You cannot provide two filter rules with equal sign that '
-                    'have the same key. E.g. `a=b and a=c` is not allowed.')
-
-            labels_filters_mapping[EQUAL][label_key] = label_values
-
-        elif 'null' in labels_filter:
-            match_null = re.match(r'(\S+) is null', labels_filter)
-            match_not_null = re.match(r'(\S+) is not null', labels_filter)
-            if match_null:
-                labels_filters_mapping.setdefault(IS_NULL, []).append(
-                    match_null.group(1).lower())
-            elif match_not_null:
-                labels_filters_mapping.setdefault(IS_NOT_NULL, []).append(
-                    match_not_null.group(1).lower())
+    parsed_filter_rules = []
+    for filter_rule in filter_rules:
+        _assert_filter_rule_structure(filter_rule)
+
+        filter_rule_key = filter_rule['key']
+        filter_rule_values = filter_rule['values']
+        filter_rule_type = filter_rule['type']
+        filter_rule_operator = filter_rule['operator']
+
+        if not isinstance(filter_rule_key, text_type):
+            raise BadFilterRule(filter_rule,
+                                'The filter rule key must be of type string')
```

Comment:
```suggestion
                                'The filter rule key must be a string')
```
if you want to make it 2 words shorter :P
(ditto list, 2 lines below)

Reply:
Done